nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -379,6 +379,26 @@ protected Document getChild(String parentId, String name) {
         }
     }
 
+    protected List<String> getChildrenIds(String parentId, boolean excludeSpecialChildren) {
+        if (isOrderable(parentId)) {
+            // TODO get only id and pos, not full state
+            // TODO state not for update
+            List<DBSDocumentState> docStates = excludeSpecialChildren
+                    ? transaction.getRegularChildrenStates(parentId)
+                    : transaction.getSpecialChildrenStates(parentId);
```

Comment:
So is that second part doing regular + special children then? If yes maybe it should be called `getAllChildrenStates()`?

Reply:
getChildrenStates() already exists and gets all children and is used in other operations than copy.
getSpecialChildrenStates() only gets the special children
getRegularChilodrenStates() only gets regular children

In this case we only want special children

Reply:
`getChildrenStates` will be renamed `getAllChildrenStates`
same for `getChildrenIds`